Source/Core/Core/PowerPC/JitArm64/JitArm64_FloatingPoint.cpp

```diff
@@ -392,3 +392,92 @@ void JitArm64::fsubx(UGeckoInstruction inst)
 
 	fpr.Unlock(V0);
 }
+
+void JitArm64::fcmpx(UGeckoInstruction inst)
+{
+	INSTRUCTION_START
+	JITDISABLE(bJITFloatingPointOff);
+
+	u32 a = inst.FA, b = inst.FB;
+	int crf = inst.CRFD;
+
+	ARM64Reg VA = fpr.R(a);
+	ARM64Reg VB = fpr.R(b);
+
+	ARM64Reg WA = gpr.GetReg();
+	ARM64Reg XA = EncodeRegTo64(WA);
+
+	FixupBranch pNaN1, pNaN2, pNaN3, pLesser, pGreater;
+	FixupBranch continue1, continue2, continue3;
+	ORR(XA, ZR, 32, 0, true);
+
+	if (a != b)
+	{
+		m_float_emit.FCMP(EncodeRegToDouble(VA), EncodeRegToDouble(VA));
+
+		// if (B != B) or (A != A), goto NaN's jump target
+		pNaN1 = B(CC_NEQ);
+
+		m_float_emit.FCMP(EncodeRegToDouble(VB), EncodeRegToDouble(VB));
+
+		pNaN2 = B(CC_NEQ);
+	}
+
+	m_float_emit.FCMP(EncodeRegToDouble(VA), EncodeRegToDouble(VB));
+
+	if (a == b)
+		pNaN3 = B(CC_NEQ);
+
+	if (a != b)
+	{
+		// if B > A goto Greater's jump target
+		pGreater = B(CC_GT);
+	}
+
+	if (a != b)
+	{
+		// if B < A, goto Lesser's jump target
+		pLesser = B(CC_MI);
+	}
+
+	ORR(XA, XA, 64 - 63, 0, true);
+	continue1 = B();
+
+	if (a != b)
```

Comment:
Problem is that conditional branches can only branch +-1MB of the current PC, and the farcode is quite a bit farther away than that. Would need some re-imagining.